### PR TITLE
Trwalke/configure await code analysis

### PR DESCRIPTION
**Pull request by trwalke**
Integration of Async Analyzers into MSAL and resolution of errors and warnings.
Associated User Story:
https://identitydivision.visualstudio.com/DevEx/_workitems/edit/307486

### CODE REVIEW

File: src/Microsoft.Identity.Client/Platforms/winrt/DispatcherTaskExtensions.cs [hunk 1]
```diff
@@ -37,18 +37,21 @@ internal static class DispatcherTaskExtensions
             Func<Task<T>> func, CoreDispatcherPriority priority = CoreDispatcherPriority.High)
         {
             var taskCompletionSource = new TaskCompletionSource<T>();
-            await dispatcher.RunAsync(priority, async () =>
+
+            async void AgileCallback()
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/Platforms/winrt/DispatcherTaskExtensions.cs]: >AgileCallback [](start = 23, length = 13)

why this name?

Comment (trwalke) [reply src/Microsoft.Identity.Client/Platforms/winrt/DispatcherTaskExtensions.cs]: @henrik-me auto suggestion by resharper. I couldn't think of anything much more meaningful given the context. perhaps TaskCallback is better?

File: src/Microsoft.Identity.Client/Platforms/winrt/DispatcherTaskExtensions.cs [hunk 2]
```diff
@@ -37,18 +37,21 @@ internal static class DispatcherTaskExtensions
             Func<Task<T>> func, CoreDispatcherPriority priority = CoreDispatcherPriority.High)
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/Platforms/winrt/DispatcherTaskExtensions.cs]: >High [](start = 89, length = 4)

does this have to be high?

File: src/Microsoft.Identity.Client/Platforms/winrt/DispatcherTaskExtensions.cs (hunk 1 above)
Comment (henrik-me) [reply src/Microsoft.Identity.Client/Platforms/winrt/DispatcherTaskExtensions.cs]: thoughts: taskCompletionDelegate/taskCompletionCallback

---
In reply to: [173580599](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/536#discussion_r173580599) [](ancestors = 173580599)

File: src/Microsoft.Identity.Client/Platforms/winrt/DispatcherTaskExtensions.cs (hunk 2 above)
Comment (trwalke) [reply src/Microsoft.Identity.Client/Platforms/winrt/DispatcherTaskExtensions.cs]: @henrik-me I do not know. I have not seen this file before. There is also a note on the documentation that says "Do not use this priority level in your app. It is reserved for system events. Using this priority can lead to the starvation of other messages, including system events."
(https://docs.microsoft.com/en-us/uwp/api/windows.ui.core.coredispatcherpriority)

@jmprieur any ideas as to why we use CoreDispatcherPriority.High?

**Pull request merged**